File: frontend/src/types/file.ts
export interface FileItem {
  id: number;
  filename: string;
  original_filename: string;
  file_size: number;
  file_type: string;
  cloudinary_public_id: string;
  cloudinary_url: string;
  category: string | null;
  folder: string | null;
  description: string | null;
  created_at: string;
  user_id: number;
}

export interface FileUploadResponse {
  id: number;
  filename: string;
  original_filename: string;
  file_size: number;
  file_type: string;
  cloudinary_public_id: string;
  cloudinary_url: string;
  category: string | null;
  folder: string | null;
  description: string | null;
  created_at: string;
}

export interface FileUpdateRequest {
  category?: string;
  folder?: string;
  description?: string;
}

File: backend/app/schemas/file.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class FileUploadResponse(BaseModel):
    id: int
    filename: str
    original_filename: str
    file_size: int
    file_type: str
    cloudinary_public_id: str
    cloudinary_url: str
    category: Optional[str]
    folder: Optional[str]
    description: Optional[str]
    created_at: datetime

    class Config:
        from_attributes = True


class FileListResponse(BaseModel):
    id: int
    filename: str
    original_filename: str
    file_size: int
    file_type: str
    cloudinary_public_id: str
    cloudinary_url: str
    category: Optional[str]
    folder: Optional[str]
    description: Optional[str]
    created_at: datetime
    user_id: int

    class Config:
        from_attributes = True


class FileUpdateRequest(BaseModel):
    category: Optional[str] = None
    folder: Optional[str] = None
    description: Optional[str] = None
